src/modules/reporting/actions/estimate-submit-market-contribute-test.js

```diff
@@ -0,0 +1,95 @@
+import { describe, it } from 'mocha'
+import { assert } from 'chai'
+import proxyquire from 'proxyquire'
+import sinon from 'sinon'
+import testState from 'test/testState'
+import configureMockStore from 'redux-mock-store'
+import thunk from 'redux-thunk'
+import { estimateSubmitMarketContribute, __RewireAPI__ as estimateSubmitMarketContributeReqireAPI } from 'modules/reporting/actions/estimate-submit-market-contribute'
+
+describe(`modules/reporting/actions/estimate-submit-market-contribute.js`, () => {
+  proxyquire.noPreserveCache().noCallThru()
+  const state = Object.assign({}, testState)
+  const mockSelectMarket = {}
+  mockSelectMarket.selectMarket = sinon.stub().returns(state.marketsData.testMarketId)
+  const mockStore = configureMockStore([thunk])
+  const store = mockStore(state)
+
+  const augurSuccess = {
+    api: {
+      Market: {
+        contribute: (options) => {
+          options.onSuccess('gasCostValue')
+        },
+      },
+    },
+  }
+
+  const augurFailed = {
+    api: {
+      Market: {
+        contribute: (options) => {
+          options.onFailed('Error')
+        },
+      },
+    },
+  }
+
+  const test = (t) => {
+    it(t.description, () => {
+      t.assertions()
+    })
+  }
+
+  after(() => {
+    estimateSubmitMarketContributeReqireAPI.__ResetDependency__('augur')
+  })
+
+  describe('empty marketId', () => {
+    test({
```

Comment:
I'm not sure what this test function buys you. Can be rewritten as:

```
  describe('empty marketId', () => {
    it(`should call the expected method`, () => {
      estimateSubmitMarketContributeReqireAPI.__Rewire__('augur', augurSuccess)
      store.dispatch(estimateSubmitMarketContribute('', 1000, (value) => {
        assert.deepEqual(value, 'Market not found', `Didn't value as expected`)
      }))
    })
  })
```

Reply:
testing error message via callback

Reply:
Sorry. I was referring to this bit:
```
  const test = (t) => {
    it(t.description, () => {
      t.assertions()
    })
  }
```

It uses roughly the same amount of space as the equivalent mocha functions.

Reply:
It's used all over the place in the unit tests, I'll convert them

Reply:
that makes sense, not getting thing out of the test method